hazelcast/src/main/java/com/hazelcast/query/impl/UnorderedIndexStore.java

```diff
@@ -62,6 +62,49 @@ Object removeInternal(Comparable value, Data recordKey) {
         return removeFunctor.invoke(value, recordKey);
     }
 
+    @Override
+    public Comparable canonicalizeScalarForHashLookup(Comparable value) {
+        return canonicalizeScalarForStorage(value);
+    }
+
+    @Override
+    public Comparable canonicalizeScalarForStorage(Comparable value) {
+        // On-heap overhead is 12 bytes for the object header, allocation
```

Comment:
Just for clarification:
- what's the difference in use of `canonicalizeScalarForHashLookup` and `canonicalizeScalarForStorage`?
- what's the effect of this comment about overhead?
- what is the reason behind the difference in behaviour of HD and non-HD and ordered vs non-ordered?

Reply:
Renamed `canonicalizeScalarForHashLookup` to `canonicalizeQueryArgumentScalar` to reflect its purpose better and added clarifying comments/javadocs.